canvas2svg.js

```diff
@@ -1148,6 +1147,29 @@
             svgImage.setAttributeNS("http://www.w3.org/1999/xlink", "xlink:href",
                 image.nodeName === "CANVAS" ? image.toDataURL() : image.getAttribute("src"));
             parent.appendChild(svgImage);
+        } else if (image.nodeName === "CANVAS") {
+            svgImage = this.__createElement("image");
+            svgImage.setAttribute("width", dw);
+            svgImage.setAttribute("height", dh);
+            svgImage.setAttribute("preserveAspectRatio", "none");
+
+            // draw canvas onto temporary canvas so that smoothing can be handled
+            canvas = this.__document.createElement("canvas");
+            canvas.width = dw;
+            canvas.height = dh;
+            var imageSmoothingEnabled = this.imageSmoothingEnabled;
```

Comment:
Or put a different way: there is never a time that Bokeh wants to render images to canvas with smoothing enabled.